### PR TITLE
Add MACsec offload feature

### PR DESCRIPTION
**Pull request by Pterosaur**
1. Add SAI_HOSTIF_ATTR_MACSEC_OFFLOAD_CAPABILITY in sai_hostif_attr
to define whether enable MACsec offload
2. Add set_hostif_macsec_offload_sa_next_pn for next packet number
 of ingress/egress setting

Signed-off-by: zegan <ganze718@gmail.com>

### CODE REVIEW

**Comment by Pterosaur** [comment]
retest this please

File: inc/saihostif.h [hunk 1]
```diff
@@ -1280,6 +1289,24 @@ typedef void (*sai_packet_event_notification_fn)(
...
+ *
+ * @param[in] hostif_id Host interface id
+ * @param[in] sci SCI value for this Secure Channel, carried in MACsec packet SecTAG.
+ * @param[in] an AN value (2-bit) for this Secure Channel, carried in MACsec packet SecTAG.
+ *               The value must be distinct from other Secure Associations for the same Secure Channel.
+ * @param[in] next_pn The next packet number of MACsec ingress/egress SA
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_set_hostif_macsec_offload_sa_next_pn_fn)(
```
Comment (kcudnik) [inline inc/saihostif.h:1304]: can this be attribute ? instead of function ?

**Comment by kcudnik** [comment]
please address compilation issues, and since you introducing new object type, you will also need to update sai_object_type_t enum

**Comment by lguohan** [comment]
@Pterosaur, please also add some doc (in md format) to describe the motivation here.

**Comment by Pterosaur** [comment]
> please address compilation issues, and since you introducing new object type, you will also need to update sai_object_type_t enum

Thanks for your suggestion, I have tried to fix the issue and add a new object type in the enum. Please review them.

**Comment by Pterosaur** [comment]
> @Pterosaur, please also add some doc (in md format) to describe the motivation here.

Thanks for your suggestion, I have added a doc for the motivation of MACsec offload interfaces. Please review it.

**Pull request closed without merging**